src/test/common/channel-control.ts

```diff
@@ -19,9 +19,9 @@ export interface ChannelControl<X, Y> {
   
     // listening
     initCompleteListener(testId: string): Promise<Y>
-    setupAndValidateListener1(channel: X, expectedContextType: string, errorMessage: string, onComplete: (ctx: Y) => void): void | Promise<void>;
-    setupAndValidateListener2(channel: X, expectedContextType: string, errorMessage: string, onComplete: (ctx: Y) => void): void | Promise<void>;
-    setupContextChecker(channel: X, expectedContextType: string, errorMessage: string, onComplete: (ctx: Y) => void): Promise<void>;
+    setupAndValidateListener1(channel: X | null, expectedContextType: string | null, errorMessage: string, onComplete: (ctx: Y) => void): void | Promise<void>;
+    setupAndValidateListener2(channel: X | null, expectedContextType: string | null, errorMessage: string, onComplete: (ctx: Y) => void): void | Promise<void>;
+    setupContextChecker(channel: X, requestedContextType: string, expectedContextType: string, errorMessage: string, onComplete: (ctx: Y) => void): Promise<void>;
```

Comment:
I think one more util function is needed here for cases where context should NOT be received (ensuring we can have a better error message than
> expected someContexttype to equal "unexpected-context"

```suggestion
    setupContextChecker(channel: X, requestedContextType: string, expectedContextType: string, errorMessage: string, onComplete: (ctx: Y) => void): Promise<void>;
```

```suggestion
    validateContextIsNotReceived = (channel: X, listenerContextType: string | null, errorMessage: string, onComplete: (ctx: Y) => void): void | Promise<void>;
    setupContextChecker(channel: X, requestedContextType: string, expectedContextType: string, errorMessage: string, onComplete: (ctx: Y) => void): Promise<void>;
```